docker-compose/docker-compose.yaml

```diff
@@ -2,23 +2,75 @@ version: '2.4'
+  #                               are set:
+  #                               - "SRC_SITE_ADDRESS": The externally accessible URL for Sourcegraph
+  #                                                     (i.e. what you type in to your browser). Example:
+  #                                                     "sourcegraph.example.com"
+  #
+  # https.custom-cert.Caddyfile: Serves Sourcegraph over HTTPS using a custom SSL certficate / key that
+  #                              you provide.
+  #                              This configuration requires that the following environment
+  #                              variable(s) are set:
+  #                              - "SRC_SITE_ADDRESS": The externally accessible URL for Sourcegraph
+  #                                                    (i.e. what you type in to your browser). Example:
+  #                                                    "sourcegraph.example.com"
+  #                              - "SRC_CERT_PATH":    The absolute path (inside the docker container) to your
+  #                                                    custom SSL certficate. Example: "/sourcegraph.pem"
+  #                                                    Note: The certficate must be bind mounted from the host machine
+  #                                                    to the Docker container.
+  #                              - "SRC_KEY_PATH":     The absolute path (inside the docker container) to your
+  #                                                    custom SSL certficate's key. Example: "/sourcegraph.key"
+  #                                                    Note: The key must be bind mounted from the host machine
+  #                                                    to the Docker container.
+  #
+  # You can swap between these configurations by setting their corresponding environment variables and
+  # editing the volume mount that mounts one of the above files into the container's /etc/caddy/Caddyfile.
+  # For example, if you wanted to use a custom certificate for https (https.custom-cert.Caddyfile) you'd do the following:
+  #   1) Set the "SRC_SITE_ADDRESS", "SRC_CERT_PATH", and "SRC_KEY_PATH" enviroment variables (and mount the local cert/key into the container).
+  #   2) Change the Caddyfile volume mount
+  #      from "- ../caddy/builtins/http.Caddyfile:/etc/caddy/Caddyfile"
+  #      to "- ../caddy/builtins/https.lets-encrypt.Caddyfile:/etc/caddy/Caddyfile"
+  #
+  # If none of these built-in configurations suit your needs, then you can create your own Caddyfile by referencing
+  # the official documentation: https://caddyserver.com/docs/caddyfile
+  caddy:
+    container_name: caddy
+    image: 'index.docker.io/caddy/caddy:alpine@sha256:b0f0a763527ca11a555bc5b5f942fa2d84ae0c90b0b60e47d0426ff884dd10ad'
     cpus: 1
     mem_limit: '1g'
+    environment:
+      - 'XDG_DATA_HOME=/caddy-storage/data'
+      - 'XDG_CONFIG_HOME=/caddy-storage/config'
```

Comment:
Q: why does Caddy use XDG_ prefix? just curious

Reply:
And what is the default value for these? Could we use that instead of /caddy-storage ?

Reply:
https://caddyserver.com/docs/conventions#data-directory

Caddy needs to keep certain files on a writeable persistent disk (certs, etc.). `XDG_DATA_HOME`, etc. are standard env vars that tell programs where to store this kind of data. Caddy's default location is to write it under some directories under  `/root` (`$HOME` in this case), but mounting a volume to the entire `/root` directory seemed a bit dirty to me (other things can write there). 

Changing both of caddy's storage locations to be subdirectories under a new folder (`/caddy-storage`) and putting only that data on the mounted volume seems like a cleaner solution to me.